controller/state.go

```diff
@@ -801,6 +800,51 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 	return &compRes, nil
 }
 
+// useDiffCache will determine if the diff should be calculated based
+// on the existing live state cache or not.
+func useDiffCache(noCache bool, manifestInfos []*apiclient.ManifestResponse, sources []v1alpha1.ApplicationSource, app *v1alpha1.Application, manifestRevisions []string, statusRefreshTimeout time.Duration) bool {
+
+	if noCache {
+		log.Debug("useDiffCache(false): noCache is true")
```

Comment:
How about logging true/false as its own field, so it's easier to process in structured logs?

Reply:
done